### ISSUE TITLE
Security group update causes: "from_port (80) and to_port (80) must both be 0 to use the 'ALL' "-1" protocol"

### ISSUE THREAD

**Issue by kennyjwilli** [pulumi/pulumi-aws#398]
I have a security group defined as follows:

```typescript
let lbSecurityGroup = new aws.ec2.SecurityGroup("fib-lb-sg", {
    namePrefix: "fib-lb-sg",
    ingress: [{
        protocol: "tcp",
        fromPort: 80,
        toPort: 80,
        cidrBlocks: ["0.0.0.0/0"]
    }],
    egress: [{
        protocol: "-1",
        fromPort: 0,
        toPort: 0,
        cidrBlocks: ["0.0.0.0/0"]
    }],
    vpcId: network.vpcId
});
```

Now I run `pulumi up` and then update the security group to look like this:

```typescript
let lbSecurityGroup = new aws.ec2.SecurityGroup("fib-lb-sg", {
    namePrefix: "fib-lb-sg",
    ingress: [{
        protocol: "-1",
        fromPort: 80,
        toPort: 80,
        cidrBlocks: ["0.0.0.0/0"]
    }],
    egress: [{
        protocol: "-1",
        fromPort: 0,
        toPort: 0,
        cidrBlocks: ["0.0.0.0/0"]
    }],
    vpcId: network.vpcId
});
```
Run `pulumi up`. This will fail due to invalid configuration. Now we fix the configuration:

```typescript
let lbSecurityGroup = new aws.ec2.SecurityGroup("fib-lb-sg", {
    namePrefix: "fib-lb-sg",
    ingress: [{
        protocol: "-1",
        fromPort: 0,
        toPort: 0,
        cidrBlocks: ["0.0.0.0/0"]
    }],
    egress: [{
        protocol: "-1",
        fromPort: 0,
        toPort: 0,
        cidrBlocks: ["0.0.0.0/0"]
    }],
    vpcId: network.vpcId
});
```
Run `pulumi up`. This will result in this error message:

```
updating urn:pulumi:fibonacci-dev::fibonacci::aws:ec2/securityGroup:SecurityGroup::fib-lb-sg: from_port (80) and to_port (80) must both be 0 to use the 'ALL' "-1" protocol!
```

Taken from a Slack thread [here](https://pulumi-community.slack.com/archives/C84L4E3N1/p1544572756260900).

**Comment by jen20** [comment]
Hi @kennyjwilli! Thanks for the straightforward test case here - I can reproduce this issue using it. I'll dig into this and come back to it shortly.

**Comment by jen20** [comment]
Furthermore, I cannot reproduce it using the same steps in Terraform alone, so this is likely to be an issue in the `pulumi-terraform` tfbridge component.

**Comment by jen20** [comment]
A further interesting note: if you apply the third step again, we update successfully.

**Comment by jen20** [comment]
This is fixed by #500 - once that is merged, I'll get a new release out. Thanks for the detailed report allowing verification of the fix, @kennyjwilli!
